contracts/logic/StakingPoolLogicV2.sol

```diff
@@ -39,24 +39,34 @@ library StakingPoolLogicV2 {
     view
     returns (uint256)
   {
-
     if (poolData.userIndex[user] == 0) {
       return 0;
     }
 
     uint256 indexDiff = getRewardIndex(poolData) - poolData.userIndex[user];
-
     uint256 balance = poolData.userPrincipal[user];
-
     uint256 result = poolData.userReward[user] + (balance * indexDiff) / 1e9;
 
     return result;
   }
 
+  function updateRewardPerSecond(StakingPoolV2.PoolData storage poolData, address user) internal {
```

Comment:
This is the most important part of the rewardPerSecond increase. 
Please review it.